zerver/migrations/0413_set_presence_enabled_false_for_user_status_away.py

```diff
@@ -0,0 +1,29 @@
+# Generated by Django 4.0.7 on 2022-09-21 15:01
+
+from django.db import migrations
+from django.db.backends.postgresql.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+AWAY = 1
+
+# Set presence_enabled to False for all users with a UserStatus.status of AWAY.
+def set_presence_enabled_false_if_user_status_away(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    UserProfile = apps.get_model("zerver", "UserProfile")
+    UserProfile.objects.filter(userstatus__status=AWAY).update(presence_enabled=False)
```

Comment:
Do we want to write a reverse migration to set the `status` to `AWAY` when disabling presence updates?

Reply:
Wouldn't that potentially update some users who had set the privacy setting [via the documentation](https://zulip.com/help/status-and-availability#disable-updating-availability) who had not done the user status `away` setting?

Reply:
Well, it won't be precise, but what I was imagining is setting `status=AWAY` if `presence_enabled=False`. 

For the mobile apps, which right now only look at `AWAY`:
* If one applies this migration and then reverses it, 100% of users lose their AWAY state. (but the `presence_enabled` state will still have the value we want, such that if one reapplies the migration soon after, it'll be as though we didn't reverse)
* With that proposal, if one applies the migration and then reverses it, users have their AWAY state restored on the reversal; this means that the mobile apps would still show users as AWAY if and only if `presence_enabled=False` ... but would not preserve the link between those two properties as users change either setting (assuming one reversed the whole PR to go back to an old version).

In either case the corner cases aren't fantastic around users changing these settings racing with this migration being applied/unapplied, I think that means it's slightly better to have a simple reverse migration of that form.

I'm also fine with just having a comment explaining why it's not important.

Reply:
No, after having time to process it, this makes sense in the context of downgrading. Because they would go back through both migrations in the series.

I was too focused on the zoomed in commit level changes and not thinking about the zoomed out big picture.